bot/cogs/LastFM.py

```diff
@@ -0,0 +1,208 @@
+    """
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(
+        name="getsong",
+        brief="a command to get info about a song, takes in song name or song name and artist, formatting for the second is songname, artist with a comma seperating the two"
+    )
+    async def getsong(self,ctx,*,args):
+      try:
+        if ', ' in args:
+          args=args.split(', ')
+          data= get_track(args[0],args[1])
+        else:
+          data = get_track(args)
+        embed=discord.Embed(title=data['name']+' by '+data['artist']['name'],url=data['url'])
+        embed.add_field(name='Artist',value=data['artist']['name'])
+        embed.add_field(name='Album',value=data['album']['title'])
+        embed.set_thumbnail(url=data['album']['image'][2]['#text'])
+        await ctx.send(embed=embed)
+      except:
+        await ctx.send('Invalid search term, try again')
+
+    @commands.command(
+      name="getlyrics",
+      brief="a command to get lyrics of a song, takes in song name or song name and artist, formatting for the second is songname, artist with a comma seperating the two"
+    )
+    async def getlyrics(self,ctx,*,args):
+      try:
+        if ', ' in args:
+          args=args.split(', ')
+          data= get_track(args[0],args[1])
+        else:
+          data = get_track(args)
+        track= urllib.parse.quote(data['name']).lower()
+        artist=urllib.parse.quote(data['artist']['name']).lower()    
+        with urllib.request.urlopen('https://api.lyrics.ovh/v1/'+artist+'/'+track) as url:
+          lyrics=json.loads(url.read().decode())['lyrics']
+        if lyrics =='':
+          lyrics ="Lyrics couldn't be found or are unavaliable at this time"
+        embed=discord.Embed(title=data['name']+' by '+data['artist']['name'],url=data['url'],description=lyrics)
+        embed.add_field(name='Artist',value=data['artist']['name'])
+        embed.add_field(name='Album',value=data['album']['title'])
+        embed.set_thumbnail(url=data['album']['image'][2]['#text'])
+        await ctx.send(embed=embed)
+      except:
+        await ctx.send('Invalid search term, try again')
+
+    @commands.command(
+      name="getalbum",
+      brief="a command to get info about a album, takes in an album name or album name and artist, formatting for the second is album name, artist with a comma seperating the two"
+    )
+    async def getalbum(self,ctx,*,args):
+      #try:
+        if ', ' in args:
+          args=args.split(', ')
+          data= get_album(args[0],args[1])
+        else:
+          data = get_album(args)
+        embed=discord.Embed(title=data['album']['name']+' by '+data['album']['artist'],url=data['album']['url'])
+        embed.add_field(name='Artist',value=data['album']['artist'])
+        embed.set_thumbnail(url=data['album']['image'][2]['#text'])
+        try:
+          embed.add_field(name='Release Data',value=data['album']['wiki']['published'],inline=False)
+        except:
```

Comment:
generic except, it also does not do anything. It declares a string with no variable or action, this should catch a specific error, it can `pass` if it purely needs to be ignored. It should raise an exception if it is not supposed to happen though.